src/l500/ac-trigger.cpp

```diff
@@ -670,12 +671,20 @@ namespace ivcam2 {
 
                     auto df = _sf.get_depth_frame();
                     auto irf = _sf.get_infrared_frame();
-                    depth_to_rgb_calibration algo( df, irf, _cf, _pcf, cal_info, cal_regs );
+
+                    depth_to_rgb_calibration algo(df, irf, _cf, _pcf, cal_info, cal_regs, [&]() {return is_processing(); });
+
                     _from_profile = algo.get_from_profile();
                     _to_profile = algo.get_to_profile();
 
-                    auto status = algo.optimize(
-                        [this]( rs2_calibration_status status ) { call_back( status ); } );
+                    rs2_calibration_status status(RS2_CALIBRATION_FAILED); // assume fail until gets a success status.
+
+                    if (is_processing()) // check if the device still exists
+                    {
+                        status = algo.optimize(
+                            [this]( rs2_calibration_status status ) { call_back( status ); } ,
+                            [&]() {return is_processing(); });
```

Comment:
I thought we provided should_continue in the ctor...

Reply:
Right, dropped it.